stdlib/camlinternalFormat.ml

```diff
@@ -1408,9 +1411,8 @@ let format_of_iconvn = function
 
 (* Generate the format_float first argument form a float_conv. *)
 let format_of_fconv fconv prec =
-  if fconv = Float_F then "%.12g" else
     let prec = abs prec in
-    let symb = char_of_fconv fconv in
+    let symb = if fconv = Float_F then 'g' else char_of_fconv fconv in
```

Comment:
Could you add a comment as to why this corner case is necessary? (Why doesn't `char_of_fconv` already return `g`?)

Reply:
Done